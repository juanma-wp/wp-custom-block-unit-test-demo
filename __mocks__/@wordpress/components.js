import React from 'react';

// Mock each of the components as you need them. Make sure to pass though all props.
export const Button = jest.fn( props => <button { ...props } /> );
export const Loading = jest.fn( props => <span { ...props } className="loading" /> );
