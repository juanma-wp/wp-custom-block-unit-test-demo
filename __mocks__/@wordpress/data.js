const mockSelectors = {
	'core': {
		getMedia: jest.fn(),
	},
};

export const select = jest.fn( storeName => mockSelectors[ storeName ] );
export const useSelect = jest.fn( fn => fn( select ) );

const mockDispatchers = {
	'core': {},
};

export const dispatch = jest.fn( storeName => mockDispatchers[ storeName ] );
export const useDispatch = jest.fn( fn => fn( dispatch ) );

beforeEach( () => {
	select.mockClear();
	useSelect.mockClear();
	dispatch.mockClear();
	useDispatch.mockClear();

	// Clear all the mock store selectors
	Object.values( mockSelectors ).forEach( store => Object.values( store ).forEach( mock => mock.mockClear() ) );

	// Clear all the mock store dispatchers
	Object.values( mockDispatchers ).forEach( store => Object.values( store ).forEach( mock => mock.mockClear() ) );
} );
