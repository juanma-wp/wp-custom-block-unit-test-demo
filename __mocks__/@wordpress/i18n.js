// Mock __, just return the untranslated string.
export const __ = jest.fn( string => string );
