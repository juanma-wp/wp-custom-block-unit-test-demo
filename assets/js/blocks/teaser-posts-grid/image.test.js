import React from 'react';
import renderer from 'react-test-renderer';
import Image from './image';
import { useSelect as mockUseSelect } from '@wordpress/data';

const mockImage = {
	id: 456,
	alt_text: 'Test Alt Text',
	source_url: 'http://example.com/cat.jpg',
	media_details: {
		sizes: {
			thumbnail: {
				source_url: 'http://example.com/cat-thumbnail.jpg',
				width: 100,
				height: 100,
			},
			full: {
				source_url: 'http://example.com/cat.jpg',
				width: 400,
				height: 300,
			},
		},
	},
};

describe( 'Teaser Posts Grid Block Image Component', () => {
	test( 'renders loading when image is not found in store.', () => {
		const rendered = renderer.create( <Image id={ 123 } /> );
		expect( rendered.toJSON() ).toMatchSnapshot();
	} );

	test( 'renders image thumbnail.', () => {
		mockUseSelect.mockImplementationOnce( () => mockImage );

		const rendered = renderer.create( <Image id={ 456 } size="thumbnail"/> );
		expect( rendered.toJSON() ).toMatchSnapshot();
	} );

	test( 'renders full size image when requested size not found.', () => {
		mockUseSelect.mockImplementationOnce( () => mockImage );

		const rendered = renderer.create( <Image id={ 456 } size="bad-size"/> );
		expect( rendered.toJSON() ).toMatchSnapshot();
	} );
} );
