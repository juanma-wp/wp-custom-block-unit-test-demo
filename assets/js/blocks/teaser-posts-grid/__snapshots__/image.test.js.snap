// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teaser Posts Grid Block Image Component renders full size image when requested size not found. 1`] = `
<img
  alt="Test Alt Text"
  height={300}
  src="http://example.com/cat.jpg"
  width={400}
/>
`;

exports[`Teaser Posts Grid Block Image Component renders image thumbnail. 1`] = `
<img
  alt="Test Alt Text"
  height={100}
  src="http://example.com/cat-thumbnail.jpg"
  width={100}
/>
`;

exports[`Teaser Posts Grid Block Image Component renders loading when image is not found in store. 1`] = `
<span
  className="loading"
/>
`;
